src/dlr_tvm.cc

```diff
@@ -224,22 +224,25 @@ void TVMModel::UseCPUAffinity(bool use) {
   }
 }
 
-bool TVMModel::HasMetadata() const { return this->metadata != nullptr; }
+bool TVMModel::HasMetadata() const { return !this->metadata.is_null(); }
 
 const char* TVMModel::GetOutputName(const int index) const {
   if (!this->HasMetadata()) {
     LOG(INFO) << "No metadata file was found!";
     return nullptr;
   }
-  if (this->metadata.count("Model") &&
-      this->metadata["Model"].count("Outputs") &&
-      index >= 0 && index < this->metadata["Model"]["Outputs"].size() &&
-      this->metadata["Model"]["Outputs"][index].count("name")) {
-    return this->metadata["Model"]["Outputs"][index]["name"]
+  try {
+    return this->metadata.at("Model")
+        .at("Outputs")
+        .at(index)
+        .at("name")
         .get_ref<const std::string&>()
         .c_str();
+  } catch (nlohmann::json::exception&) {
```

Comment:
Could you test that this exception is caught when index is out of range of the outputs in the metadat afile?

Reply:
I already tested accessing index 0 when Outputs array was empty

Reply:
I  was able to repro seg fault - it happens in case model has 4 outputs but metadata has only 3 items in Output Array
```
output_names: out0 (index: 0), out1 (index: 1), out2 (index: 2), [23:58:53] /root/workplace/neo-ai-dlr/src/dlr_tvm.cc:238: Output node with index 3 not found in metadata file
Segmentation fault (core dumped)
```

Reply:
It was a bug in `dlr.cc` - `GetDLROutputName`:
```
CHECK(name != nullptr) 
// instead of 
CHECK(*name != nullptr)
```